x/reward/keeper/claim_reward_distribution_test.go

```diff
@@ -0,0 +1,198 @@
+	rewardDistribution := types.NewClaimPeriodRewardDistribution(
+		1,
+		2,
+		sdk.NewInt64Coin("nhash", 100000),
+		sdk.NewInt64Coin("nhash", 200000),
+		3,
+		true,
+	)
+	suite.app.RewardKeeper.SetClaimPeriodRewardDistribution(suite.ctx, rewardDistribution)
+	retrieved, err := suite.app.RewardKeeper.GetClaimPeriodRewardDistribution(suite.ctx, 1, 2)
+	suite.Assert().NoError(err)
+	suite.Assert().Equal(rewardDistribution.ClaimPeriodId, retrieved.ClaimPeriodId, "claim period id must match")
+	suite.Assert().Equal(rewardDistribution.RewardProgramId, retrieved.RewardProgramId, "reward program id must match")
+	suite.Assert().Equal(rewardDistribution.RewardsPool, retrieved.RewardsPool, "rewards pool must match")
+	suite.Assert().Equal(rewardDistribution.TotalRewardsPoolForClaimPeriod, retrieved.TotalRewardsPoolForClaimPeriod, "total rewards pool must match")
+	suite.Assert().Equal(rewardDistribution.TotalShares, retrieved.TotalShares, "total shares must match")
+	suite.Assert().Equal(rewardDistribution.ClaimPeriodEnded, retrieved.ClaimPeriodEnded, "claim period ended must match")
+
+	retrieved, err = suite.app.RewardKeeper.GetClaimPeriodRewardDistribution(suite.ctx, 1, 99)
+	suite.Assert().NoError(err)
+	suite.Assert().Equal(uint64(0), retrieved.RewardProgramId, "reward program id must be invalid")
+}
+
+func (suite *KeeperTestSuite) TestIterateClaimPeriodRewardDistributions() {
+	tests := []struct {
+		name          string
+		distributions []types.ClaimPeriodRewardDistribution
+		halt          bool
+		counter       int
+	}{
+		{
+			"valid - can handle empty distributions",
+			[]types.ClaimPeriodRewardDistribution{},
+			false,
+			0,
+		},
+		{
+			"valid - can handle one distribution",
+			[]types.ClaimPeriodRewardDistribution{
+				types.NewClaimPeriodRewardDistribution(1, 1, sdk.NewInt64Coin("nhash", 100), sdk.NewInt64Coin("nhash", 100), 1, false),
+			},
+			false,
+			1,
+		},
+		{
+			"valid - can handle multiple distributions",
+			[]types.ClaimPeriodRewardDistribution{
+				types.NewClaimPeriodRewardDistribution(1, 1, sdk.NewInt64Coin("nhash", 100), sdk.NewInt64Coin("nhash", 100), 1, false),
+				types.NewClaimPeriodRewardDistribution(1, 2, sdk.NewInt64Coin("nhash", 100), sdk.NewInt64Coin("nhash", 100), 1, false),
+				types.NewClaimPeriodRewardDistribution(1, 3, sdk.NewInt64Coin("nhash", 100), sdk.NewInt64Coin("nhash", 100), 1, false),
+			},
+			false,
+			3,
+		},
+		{
+			"valid - can handle halting",
+			[]types.ClaimPeriodRewardDistribution{
+				types.NewClaimPeriodRewardDistribution(1, 1, sdk.NewInt64Coin("nhash", 100), sdk.NewInt64Coin("nhash", 100), 1, false),
+				types.NewClaimPeriodRewardDistribution(1, 2, sdk.NewInt64Coin("nhash", 100), sdk.NewInt64Coin("nhash", 100), 1, false),
+				types.NewClaimPeriodRewardDistribution(1, 3, sdk.NewInt64Coin("nhash", 100), sdk.NewInt64Coin("nhash", 100), 1, false),
+			},
+			true,
+			1,
+		},
+	}
+
+	for _, tt := range tests {
```

Comment:
Nit: Usually we use `tc` for this variable name.

Reply:
Good catch, will change.